src/datumaro/components/algorithms/rise.py

```diff
@@ -34,186 +25,104 @@ class RISE:
     def __init__(
         self,
         model,
-        max_samples=None,
-        mask_width=7,
-        mask_height=7,
-        prob=0.5,
-        iou_thresh=0.9,
-        nms_thresh=0.0,
-        det_conf_thresh=0.0,
-        batch_size=1,
+        num_masks: int = 100,
+        mask_size: int = 7,
+        prob: float = 0.5,
+        batch_size: int = 1,
     ):
+        assert prob >= 0 and prob <= 1
         self.model = model
-        self.max_samples = max_samples
-        self.mask_height = mask_height
-        self.mask_width = mask_width
+        self.num_masks = num_masks
+        self.mask_size = mask_size
         self.prob = prob
-        self.iou_thresh = iou_thresh
-        self.nms_thresh = nms_thresh
-        self.det_conf_thresh = det_conf_thresh
         self.batch_size = batch_size
 
-    @staticmethod
-    def split_outputs(annotations):
-        labels = []
-        bboxes = []
-        for r in annotations:
-            if r.type is AnnotationType.label:
-                labels.append(r)
-            elif r.type is AnnotationType.bbox:
-                bboxes.append(r)
-        return labels, bboxes
-
-    def normalize_hmaps(self, heatmaps, counts):
-        eps = np.finfo(heatmaps.dtype).eps
-        mhmaps = _flatmatvec(heatmaps)
-        mhmaps /= _expand(counts * self.prob + eps)
-        mhmaps -= _expand(np.min(mhmaps, axis=1))
-        mhmaps /= _expand(np.max(mhmaps, axis=1) + eps)
-        return np.reshape(mhmaps, heatmaps.shape)
+    def normalize_saliency(self, saliency):
+        normalized_saliency = np.empty_like(saliency)
+        for idx, sal in enumerate(saliency):
+            normalized_saliency[idx, ...] = (sal - np.min(sal)) / (np.max(sal) - np.min(sal))
+        return normalized_saliency
```

Comment:
```suggestion
        return (saliency - np.min(saliency)) / (np.max(saliency) - np.min(saliency))
```
I think this could be replaced like this through vectorized operation.

Reply:
thank you for the feedback, but the original code is to normalize each saliency map per class_idx, while your suggestion is to normalize each saliency map from global max and min. So I expect the results will be different.